packages/git-utils/src/main.js

```diff
@@ -11,25 +11,57 @@ const getGitUtils = async function(
   // istanbul ignore next
   { base, cwd = getCwd() } = {},
 ) {
-  const properties = await addProperties(base, cwd)
-  const gitUtils = addMethods(properties)
-  return gitUtils
+  try {
+    const baseA = await getBase(base, cwd)
+    const properties = await getProperties(baseA, cwd)
+    const gitUtils = addMethods(properties)
+    return gitUtils
+  } catch (error) {
+    return { error: error.stack }
+  }
 }
 
-const addProperties = async function(base, cwd) {
-  const baseA = await getBase(base, cwd)
+const getProperties = async function(base, cwd) {
   const [{ modifiedFiles, createdFiles, deletedFiles }, commits, linesOfCode] = await Promise.all([
-    getDiffFiles(baseA, cwd),
-    getCommits(baseA, cwd),
-    getLinesOfCode(baseA, cwd),
+    getDiffFiles(base, cwd),
+    getCommits(base, cwd),
+    getLinesOfCode(base, cwd),
   ])
   return { modifiedFiles, createdFiles, deletedFiles, commits, linesOfCode }
 }
 
 const addMethods = function(properties) {
+  // During initialization, the git utility is not built yet since it is a Proxy
+  // and the utility must be sent across processes, i.e. JSON-serializable.
+  // Instead we only keep the error stack and build the Proxy now.
+  if (properties.error !== undefined) {
+    return getFakeGitUtils(properties)
+  }
+
   const fileMatchA = fileMatch.bind(null, properties)
   return { ...properties, fileMatch: fileMatchA }
 }
 
+// Initialization might fail for many reasons, including the repository not
+// having a `.git` directory or some `git` commands failing.
+// However initialization happens at the beginning of the build. At that stage,
+// we do not know whether the `git` utility will be used yet. In those cases,
+// we don't want to report any errors since the user might not use the utility.
+// However we still want to report errors when the user does use the utility.
+// We achieve this by using a Proxy.
+const getFakeGitUtils = function({ error }) {
+  return new Proxy(
+    // We define those so that `Object.keys()` and similar methods still work
+    { modifiedFiles: [], createdFiles: [], deletedFiles: [], commits: [], linesOfCode: 0, fileMatch() {} },
```

Comment:
Looks good thanks for implementing this 😃

Can we return the null values from `fileMatch` too?

Null return would look like:

```
{
  modified: [ ],
  created: [ ],
  deleted: [ ],
  edited: [ ]
}
```

Reply:
:+1: Fixed in b06c6589cd748d42c85ddf8b4fa6f131bd51750a